app/assets/javascripts/editor.js

```diff
@@ -23,117 +26,110 @@ $E = {
     //   1. /wiki/new
     //   2. /wiki/{wiki name}/edit
     //   3. /features/new
-    const legacyEditorPath = RegExp(/\/(wiki|features)(\/[^\/]+\/edit|\/new)/);
-    return !legacyEditorPath.test(url); // if we're not on one of these pages, we are using the rich-text editor.
-  },
+    const singleFormPath = RegExp(/\/(wiki|features)(\/[^\/]+\/edit|\/new)/);
+    return singleFormPath.test(url);
+  }
   // wraps currently selected text in textarea with strings a and b
-  wrap: function(a, b, args) {
-    // we only refresh $E's values if we are on a page using the rich-text editor (most pages).
-    // the legacy editor pages only have one editor form, unlike pages with multiple comments.
-    if (this.isRichTextEditor(window.location.pathname)) { this.refresh(); }
-    var len = $E.textarea.val().length;
-    var start = $E.textarea[0].selectionStart;
-    var end = $E.textarea[0].selectionEnd;
-    const fallbackParameterExists = args && args['fallback'];
-    const newlineParameterExists = args && args['newline'];
-    var sel = fallbackParameterExists ? args['fallback'] : $E.textarea.val().substring(start, end); // // fallback if nothing has been selected, and we're simply dealing with an insertion point
-    var replace = a + sel + b;
-    if (newlineParameterExists) {
-      replace = replace + "\n\n";
-    }
-    if (newlineParameterExists && $E.textarea[0].selectionStart > 0) { 
-      replace = "\n" + replace; 
-    }
-    $E.textarea.val($E.textarea.val().substring(0,start) + replace + $E.textarea.val().substring(end,len));
-  },
-  bold: function() {
+  wrap(a, b, newlineDesired = false, fallback) {
+    // we only refresh $E's values if we are on a page with multiple comments
+    if (!this.isSingleFormPage(window.location.pathname)) { this.refresh(); }
+
+    const selectionStart = $E.textarea[0].selectionStart;
+    const selectionEnd = $E.textarea[0].selectionEnd;
+    const selection = fallback || $E.textarea.val().substring(selectionStart, selectionEnd); // fallback if nothing has been selected, and we're simply dealing with an insertion point
+
+    let newText = a + selection + b; // ie. ** + selection + ** (wrapping selection in bold)
+    if (newlineDesired) { newText = newText + "\n\n"; }
+    const selectionStartsMidText = $E.textarea[0].selectionStart > 0;
+    if (newlineDesired && selectionStartsMidText) { newText = "\n" + newText; }
+
+    const textLength = $E.textarea.val().length;
+    const textBeforeSelection = $E.textarea.val().substring(0, selectionStart);
+    const textAfterSelection = $E.textarea.val().substring(selectionEnd, textLength);
+    $E.textarea.val(textBeforeSelection + newText + textAfterSelection);
+  }
+  bold() {
     $E.wrap('**','**')
-  },
-  italic: function() {
+  }
+  italic() {
     $E.wrap('_','_')
-  },
-  link: function(uri) {
+  }
+  link(uri) {
     uri = prompt('Enter a URL');
     if (uri === null) { uri = ""; }
```

Comment:
```suggestion
    if (!!uri) { uri = ""; }
```

Reply:
Interesting, I've never heard of `!!` to check for `null`!

Reply:
Hey it is single !

Reply:
So !! gives false for undefined, null, 0 and “”

Reply:
Okay! I think it makes sense to check for `undefined` too, so will commit this. Thank you!